web/src/components/Dialogs/CollectionDialog.tsx

```diff
@@ -0,0 +1,58 @@
+import Card from "../Surfaces/Card";
+import ImageCard from "../Surfaces/ImageCard";
+import StyledDialog from "./StyledDialog";
+
+interface Props {
+  src: string;
+  name: string;
+  authorName: string;
+  authorAvatar: string;
+  posts: {
+    id: number;
+    src: string;
+    prompt: string;
+    likes: number;
+    authorName: string;
+    authorAvatar: string;
+  }[];
```

Comment:
I would suggest extracting the `Post` type to a type of its own, that way you can reuse the type definition in other files.